lib/src/qonversion.dart

```diff
@@ -56,6 +57,65 @@ class Qonversion {
     return QMapper.productsFromJson(rawResult);
   }
 
+  /// Starts a process of purchasing product with [productId].
+  ///
+  /// Throws `QPurchaseException` in case of error in purchase flow.
+  static Future<Map<String, QPermission>> purchase(String productId) async {
+    final rawResult = await _channel
+        .invokeMethod(Constants.mPurchase, {Constants.kProductId: productId});
+
+    final resultMap = Map<String, dynamic>.from(rawResult);
+
+    final error = resultMap[Constants.kError];
+    if (error != null) {
+      throw QPurchaseException(
+        error,
+        isUserCancelled: resultMap[Constants.kIsCancelled] ?? false,
+      );
+    }
+
+    return QMapper.permissionsFromJson(resultMap[Constants.kPermissions]);
+  }
+
+  /// Android only. Returns `null` if called on iOS.
+  ///
+  /// Upgrading, downgrading, or changing a subscription on Google Play Store requires calling updatePurchase() function.
+  ///
+  /// See [Google Play Documentation](https://developer.android.com/google/play/billing/subscriptions#upgrade-downgrade) for more details.
+  static Future<Map<String, QPermission>> updatePurchase({
```

Comment:
Looks like you forgot to add an extended version of this function which contains `prorationMode` optional param